server/pull_request_handler.go

```diff
@@ -157,29 +181,45 @@ func (s *Server) handlePullRequestEvent(ctx context.Context, event *PullRequestE
 		}
 	}
 
-	s.checkPullRequestForChanges(ctx, pr)
+	changed, err := s.checkPullRequestForChanges(ctx, pr)
+	if err != nil {
+		mlog.Error("Could not check changes for PR", mlog.Err(err))
+	} else if changed {
+		mlog.Info("pr has changes", mlog.Int("pr", pr.Number))
+	}
 }
 
-func (s *Server) checkPullRequestForChanges(ctx context.Context, pr *model.PullRequest) {
+func pullRequestEventFromJSON(data io.Reader) (*pullRequestEvent, error) {
+	decoder := json.NewDecoder(data)
+	var event pullRequestEvent
+	if err := decoder.Decode(&event); err != nil {
+		return nil, err
+	}
+
+	return &event, nil
+}
+
+func (s *Server) checkPullRequestForChanges(ctx context.Context, pr *model.PullRequest) (bool, error) {
 	oldPr, err := s.Store.PullRequest().Get(pr.RepoOwner, pr.RepoName, pr.Number)
 	if err != nil {
-		mlog.Error(err.Error())
-		return
+		return false, err
 	}
 
 	if oldPr == nil {
 		if _, err := s.Store.PullRequest().Save(pr); err != nil {
-			mlog.Error(err.Error())
+			mlog.Error("could not save PR", mlog.Err(err))
 		}
 
 		for _, label := range pr.Labels {
-			s.handlePRLabeled(ctx, pr, label)
+			if err := s.handlePRLabeled(ctx, pr, label); err != nil {
+				mlog.Error("could not handle labeled event", mlog.Err(err))
+			}
 		}
 
-		return
+		return false, nil
 	}
 
-	prHasChanges := false
+	var prHasChanges bool
 
 	for _, label := range pr.Labels {
```

Comment:
Maybe we can extract this part of the code and create a new function that receives two string slices and returns the diff between them so we can reduce the amount of boilerplate here. What do you think?

Reply:
Yeah, makes sense to avoid repetition.